### PR TITLE
Enable mtls internal; add tests for it

### PR DESCRIPTION
**Pull request by moosecanswim**
Enables mtls for internal services (manual-mtls) and add tests that will ensure these stay in a good state.

[sc-32689]

Configuration Options:

| edge      | internal  | defaults.edge.enable_tls | defaults.edge.require_client_certs | defaults.core_internal_tls_certs.enable | defaults.core_internal_tls_certs.require_client_certs | configs.spire |
| --------- | --------- | ------------------------ | ---------------------------------- | --------------------------------------- | ----------------------------------------------------- | ------------- |
| plaintext | plaintext | false                    | false                              | false                                   | false                                                 | false         |
| plaintext | spire     | false                    | false                              | false                                   | false                                                 | true          |
| tls       | plaintext | true                     | false                              | false                                   | false                                                 | false         |
| tls       | tls       | true                     | false                              | true                                    | false                                                 | false         |
| tls       | mtls      | true                     | false                              | true                                    | true                                                  | false         |
| tls       | spire     | true                     | false                              | false                                   | false                                                 | true          |
| mtls      | plaintext | true                     | true                               | false                                   | false                                                 | false         |
| mtls      | tls       | true                     | true                               | true                                    | false                                                 | false         |
| mtls      | mtls      | true                     | true                               | true                                    | true                                                  | false         |
| mtls      | spire     | true                     | true                               | false                                   | false                                                 | true          |

> External prometheus tls, tls/mtls edge and keycloak, and observables -> elasticsearch have not been tested

> all tls and mtls tests require you to create your own secret

### CODE REVIEW

File: inputs.cue [hunk 1]
```diff
@@ -160,12 +160,13 @@ defaults: {
...
 		// The value should not need to be changed.
 		key: "edge"
-		// edge.enable_tls toggles internal mtls connections between greymatter core components
-		// by default the internal and external secrets are the same however if you want to
-		// have different certs for the ingress and internal connections (this is the case for prod)
-		// then you will need to add those certs to another secret and specity that
-		// below at defaults.core_internal_tls_certs.cert_secret.
+		// To enable tls on the edge set edge.enable_tls to true.
+		// This config also toggles internal tls to be enabled.  That behavior can be changed 
+		// by setting the toggle defaults.internal.core_internal_tls_certs.enable.
```
Comment (tilleryd) [inline inputs.cue:165]:
```suggestion
		// To enable TLS on the edge set edge.enable_tls to true.
		// This config also toggles enables internal TLS across sidecars.  That behavior can be changed 
		// by setting the toggle defaults.internal.core_internal_tls_certs.enable to true.
```

File: inputs.cue [hunk 2]
```diff
@@ -160,12 +160,13 @@ defaults: {
...
 		enable_tls:  bool | *false @tag(edge_enable_tls,type=bool)
+		// To enable mtls on the edge edge.require_client_certs should be set to true in addition to enable_tls.
+		// This  conig also toggles internal mtls to be enabled.  That behavior can be changed
+		// by setting the toggle defaults.internal.core_internal_tls_certs.require_client_certs.
```
Comment (tilleryd) [inline inputs.cue:169]:
```suggestion
		// To enable mTLS on the edge, edge.require_client_certs should be set to true in addition to edge.enable_tls.
		// This config also toggles enables internal TLS across sidecars.  That behavior can be changed
		// by setting the toggle defaults.internal.core_internal_tls_certs.require_client_certs to true.
```

File: inputs.cue [hunk 3]
```diff
@@ -234,9 +235,14 @@ defaults: {
 	}
 
 	core_internal_tls_certs: {
+		// Enables internal tls (requires defaults.edge.enable_tls == true)
```
Comment (tilleryd) [inline inputs.cue:238]:
```suggestion
		// Enables internal mTLS (requires defaults.edge.enable_tls == true)
```

File: inputs.cue [hunk 4]
```diff
@@ -234,9 +235,14 @@ defaults: {
...
 		enable: bool | *defaults.edge.enable_tls @tag(internal_enable_tls,type=bool)
-		// use npe cert for internal mtls
+		// Enables internal mtls (requires: defaults.edge.enable_tls == true && defaults.core_internal_tls_certs.enable == true )
```
Comment (tilleryd) [inline inputs.cue:240]:
```suggestion
		// Enables internal mTLS (requires: defaults.edge.enable_tls == true && defaults.core_internal_tls_certs.enable == true )
```

File: inputs.cue [hunk 5]
```diff
@@ -234,9 +235,14 @@ defaults: {
...
+		require_client_certs: bool | *defaults.edge.require_client_certs @tag(internal_require_client_certs, type=bool)
 		// Name of kubernetes secret to be mounted
+		// By default the same secret for external tls/mtls will be used for internal tls/mtls.
+		// Different certs can be used by specifying a different
+		// secret name in defaults.core_internal_tls_certs.cert_secret.
```
Comment (tilleryd) [inline inputs.cue:245]:
```suggestion
		// By default the same secret for external TLS/mTLS will be used for internal TLS/mTLS.
		// Different certs can be used by specifying a different
		// secret name in defaults.core_internal_tls_certs.cert_secret.
```

**Comment by tilleryd** [comment]
Test are green, I'm inclined to approve this. I'll wait for others to review as well.

**Review by dontlaugh — APPROVED** [review]

**Pull request merged**